src/olympia/reviewers/tests/test_models.py

```diff
@@ -1241,6 +1242,29 @@ def test_check_should_be_delayed(self):
             AutoApprovalSummary.check_has_auto_approval_disabled(self.version) is False
         )
 
+    def test_check_should_be_delayed_dynamic(self):
+        # The delay defaults to 24 hours (see test above) but can be configured
+        # by admins.
+        set_config('INITIAL_AUTO_APPROVAL_DELAY_FOR_LISTED', 666)
+        # Delete current_version, making self.version the first listed version
+        # submitted and add-on creation date recent.
+        self.addon.current_version.delete()
+        self.addon.update(created=datetime.now())
+        self.addon.update_status()
+        assert AutoApprovalSummary.check_should_be_delayed(self.version) is True
+
+        self.version.update(created=datetime.now() - timedelta(seconds=660))
+        assert AutoApprovalSummary.check_should_be_delayed(self.version) is True
+        self.version.update(created=datetime.now() - timedelta(seconds=667))
+        assert AutoApprovalSummary.check_should_be_delayed(self.version) is False
+
+        # Goes back to 24 hours if the value is invalid.
+        set_config('INITIAL_AUTO_APPROVAL_DELAY_FOR_LISTED', 'nonsense')
```

Comment:
Do we not validate `set_config` for data type?

Reply:
No, it can be anything (there is no schema or anything). It's up to the caller that grabs the value (though `get_config()`) to use `int_value` or `json_value` if that's what they are expecting (and if invalid we log that and fall back to the default)